src/Database/LMDB/Simple/Cursor.hs

```diff
@@ -114,13 +116,20 @@ data PeekPoke k v = PeekPoke {
 -- We perform lower-level cursor operations in @'IO'@, while the @'ReaderT'@
 -- passes along a @'CursorEnv'@ that contains all the information we need to
 -- perform the lower-level cursor operations.
-newtype CursorM k v a = CursorM {unCursorM :: ReaderT (CursorEnv k v) IO a}
+newtype CursorM k v mode a = CursorM {unCursorM :: ReaderT (CursorEnv k v) IO a}
```

Comment:
Where do we use `ReadOnly`? I could not find an occurrence in this PR :thinking:

Reply:
Something like ...
```haskell
cgetFirst :: CursorConstraints m k v ReadOnly => m ReadOnly (Maybe (k, v))
cgetFirst = cgetG MDB_FIRST
```
... would prevent us from using `cgetFirst` in a `ReadWrite` transaction/environment. Operations that can run in both modes will typically be polymorphic in `mode`, and operations that can only run in read-write mode will specify `ReadWrite`. There's no cursor operations (that I can think of currently) that should not also be able to run in read-write mode, but if it were the case then I would specify `ReadOnly`